pkg/server/plugin/upstreamca/awssecret/awslib_fake_test.go

```diff
@@ -0,0 +1,53 @@
+package awssecret
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+type fakeSecretsManagerClient struct {
+	secretsmanageriface.SecretsManagerAPI
+
+	storage map[string]string
+}
+
+func (sm *fakeSecretsManagerClient) GetSecretValueWithContext(ctx aws.Context, input *secretsmanager.GetSecretValueInput, opt ...request.Option) (*secretsmanager.GetSecretValueOutput, error) {
+
+	if value, ok := sm.storage[*input.SecretId]; ok {
+		return &secretsmanager.GetSecretValueOutput{
+			ARN:          input.SecretId,
+			SecretString: &value,
+		}, nil
+	} else {
+		return &secretsmanager.GetSecretValueOutput{
+			ARN:          nil,
+			SecretString: nil,
+		}, fmt.Errorf("Not found")
+	}
+}
+
+func newFakeSecretsManagerClient(config *AWSSecretConfiguration, region string) (secretsmanageriface.SecretsManagerAPI, error) {
```

Comment:
consider replacing secretsmanageriface.SecretsManagerAPI with the interface I talked about above.